File: sampat_industries/public/js/doctype_plugin/BOM/bom.py
# Copyright (c) 2024, Niraj and contributors
# For license information, please see license.txt

import frappe
from frappe.model.document import Document

class SampatEmployeeAdvances(Document):
	pass

@frappe.whitelist()
def get_alternate_uom_from_item_master(item_code):
    item_alternate_uom = frappe.db.get_value('Item', {'item_code': item_code}, ['alternate_uom', 'conversion_factor'], as_dict=1)
    
    return item_alternate_uom

File: sampat_industries/public/js/doctype_plugin/BOM/bom.js
frappe.ui.form.on("BOM", {
  // refresh: function (frm) {
  //   console.log("ACCESSING BOM");
  // },
});

frappe.ui.form.on("BOM Item", {
  item_code: function (frm, cdt, cdn) {
    const child = locals[cdt][cdn];
    let item_code = child.item_code;
    frappe
      .call({
        method:
          "sampat_industries.public.js.doctype_plugin.BOM.bom.get_alternate_uom_from_item_master",
        args: { item_code },
      })
      .done((r) => {
        console.log(r.message);
        let alternate_uom = r.message.alternate_uom;
        let conversion_factor = 1 / r.message.conversion_factor;

        setTimeout(() => {
          frappe.model.set_value(cdt, cdn, "uom", alternate_uom);
        }, 1000);
        setTimeout(() => {
          frappe.model.set_value(
            cdt,
            cdn,
            "conversion_factor",
            conversion_factor
          );
        }, 2000);
      });
  },
});
